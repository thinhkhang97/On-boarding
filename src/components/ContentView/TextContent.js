// @flow
import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import Title from '../Title/Title';
import Description from '../Description/Description';
import MixTitle from '../Title/MixTitle';

type Props = {
  mix?: boolean,
  title?: string,
  description?: string,
  mixData?: {
    leftTitle: string,
    leftSubTitle: string,
    rightTitle: string,
    rightSubTitle: string,
  },
  mixTitleStyle?: View.propTypes.style,
  mixSubTitleStyle?: View.propTypes.style,
  titleStyle?: View.propTypes.style,
};

class TextContent extends React.Component<Props> {
  static defaultProps = {
    mix: false,
    mixData: {
      leftTitle: '',
      leftSubTitle: '',
      rightTitle: '',
      rightSubTitle: '',
    },
    title: '',
    description: '',
  };
  render() {
    const {
      mix,
      title,
      description,
      mixData,
      mixTitleStyle,
      mixSubTitleStyle,
      titleStyle,
    } = this.props;
    return (
      <View style={styles.container}>
        {mix ? (
          <View style={styles.titleContainer}>
            <MixTitle
              title={mixData.leftTitle}
              subTitle={mixData.leftSubTitle}
              titleStyle={mixTitleStyle}
              subTitleStyle={mixSubTitleStyle}
            />
            <MixTitle
              title={mixData.rightTitle}
              subTitle={mixData.rightSubTitle}
              titleStyle={mixTitleStyle}
              subTitleStyle={mixSubTitleStyle}
            />
          </View>
        ) : (
          <Title style={[titleStyle, { textAlign: 'center' }]}>{title}</Title>
        )}

        <Description style={styles.descriptionContainer}>{description}</Description>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: { flex: 1, justifyContent: 'space-around' },
  titleContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  descriptionContainer: {
    textAlign: 'center',
    paddingLeft: 39,
    paddingRight: 36,
  },
});

export default TextContent;
