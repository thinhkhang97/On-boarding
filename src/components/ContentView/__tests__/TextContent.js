import * as React from 'react';
import renderer from 'react-test-renderer';
import TextContent from '../TextContent';

test('Render avatar view correctly', ()=>{
    const testTextContent = renderer.create(<TextContent/>).toJSON();
    expect(testTextContent).toMatchSnapshot();
})