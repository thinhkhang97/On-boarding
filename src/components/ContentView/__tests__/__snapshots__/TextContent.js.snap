// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render avatar view correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 32,
          "fontWeight": "bold",
        },
        Array [
          undefined,
          Object {
            "textAlign": "center",
          },
        ],
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#7f8da2",
          "fontSize": 16,
        },
        Object {
          "paddingLeft": 39,
          "paddingRight": 36,
          "textAlign": "center",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
